### PR TITLE
fix: override journey template autocomplete filter to trim

### PR DESCRIPTION
**Pull request by jaycnz**
## Summary by CodeRabbit

* **Bug Fixes**
  * Tab navigation now syncs correctly with the URL and reliably switches between Active, Archived and Trashed views.

* **Improvements**
  * Tag filtering trims extra whitespace, treats inputs of only spaces as showing all options, and performs case-insensitive matching for more accurate results.

* **Tests**
  * Added comprehensive tests covering whitespace handling and filtering edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by edmonday — CHANGES_REQUESTED** [review]

File: libs/journeys/ui/src/components/TemplateGallery/TagsFilter/TagsFilter.tsx [hunk 1]
```diff
@@ -87,6 +87,19 @@ export function TagsFilter({
...
+    const trimmedInput = state.inputValue.trim().toLowerCase()
+    if (trimmedInput === '') {
+      return options
+    }
+
+    return options.filter((option) => {
+      const label = option.name.find(({ primary }) => primary)?.value ?? ''
+      return label.toLowerCase().includes(trimmedInput)
+    })
+  }
```
[the remark this answers is not in the document]
Comment (jaycnz) [reply libs/journeys/ui/src/components/TemplateGallery/TagsFilter/TagsFilter.tsx]: manual filtering now replaced with autocomplete's createFilterOptions
https://mui.com/material-ui/react-autocomplete/#createfilteroptions-config-filteroptions
